src/Termonad/Types.hs

```diff
@@ -59,28 +69,36 @@ instance Show TMTerm where
       showsPrec (d + 1) (hashUnique unique) .
       showString "}"
 
--- | A container that holds everything in a given terminal window.  The 'term'
--- in the 'TMTerm' is inside the 'tmNotebookTabTermContainer' 'ScrolledWindow'.
--- The notebook tab 'Label' is also available.
+-- | A container that holds everything in a given notebook tab.  Each tab is
+-- split in two terminals, one on the left and one on the right. The notebook
+-- tab 'Label' is also available.
 data TMNotebookTab = TMNotebookTab
-  { tmNotebookTabTermContainer :: !ScrolledWindow
-    -- ^ The 'ScrolledWindow' holding the VTE 'Terminal'.
-  , tmNotebookTabTerm :: !TMTerm
-    -- ^ The 'Terminal' insidie the 'ScrolledWindow'.
+  { tmNotebookTabPaned :: !Paned
+    -- ^ The 'Paned' holding the two 'TMTerm's.
+  , tmNotebookTabLeftTerm :: !TMTerm
+    -- ^ The left 'TMTerm'.
+  , tmNotebookTabRightTerm :: !TMTerm
+    -- ^ The right 'TMTerm'.
+  , tmNotebookTabFocusIsOnLeft :: !Bool
```

Comment:
You have a bunch of good questions here, and I don't really have any good answers.

When hearing "Panes", my first thought is `vim`, since that's probably the program I use the most that has a Pane-like feature.  Vim has a concept of vertical panes vs horizontal panes.  Each set of vertical panes (or horizontal panes) allows you to add arbitrarily many vertical panes (or horizontal panes).  If you try to add a horizontal pane to a set of vertical panes, you start a new set of horizontal panes that exists in a single vertical pane.  So in this case, I think a `FocusTree` data-type would be appropriate, where each layer of the tree has a different orientation.

> Maybe I instead need a FocusTree equivalent to a FocusList, that is, a Tree whose focus is a path to a leaf?

This was my first thought.  I'd be happy to give you commit rights to [`focus-list`](https://hackage.haskell.org/package/focuslist), or possibly instead create a separate package for some sort of FocusTree datatype.

Although, I guess it would be better to have _some_ implementation of Panes in Termonad, even if it is not ideal.  Creating a whole new FocusTree datatype might be a lot of work.  We could always improve/generalize it at some point in the future.

-------------------------------------------------------------

If I was going to try to implement panes in Termonad, the first thing I would do is probably take a look at how a few programs that use panes work, including what sorts of operations they provide for panes.  The programs I personally use that have panes are:

- XMonad
- Vim
- Emacs

Of these, I think XMonad might be the only program that has a significantly different handling for panes.

A couple questions that would be nice to answer for each of the above programs:

- Do panes seem to get held in a binary tree, with each pane individually able to be split?  Or are panes held in more of a forest (like [`Tree`](https://hackage.haskell.org/package/containers-0.6.5.1/docs/Data-Tree.html#t:Tree))?  How are panes automatically laid-out / sized?
- What operations are available for re-arranging panes?  What are the default key-shortcuts for these operations?
- What operations are available for re-sizing panes?  What are the default key-shortcuts for these operations?

Reply:
Let's answer those questions in the discussion for #205, where @craigem [already answered the key-shortcuts question](https://github.com/cdepillabout/termonad/issues/205#issuecomment-964828382) for a few applications.